opte/src/engine/packet.rs

```diff
@@ -2550,7 +2519,40 @@ mod test {
         let tcp_bytes = tcp.as_bytes();
         assert!(matches!(
             wtr.write(&tcp_bytes[0..12]),
-            Err(WriteError::NotEnoughBytes { available: 8, needed: 12 })
+            Err(WriteError::NotEnoughBytes { available: 8, needed: 12, .. })
+        ));
+    }
+
+    // Verify that modification that starts past wptr produces an
+    // OutOfRange error.
+    #[test]
+    fn out_of_range() {
+        let mp1 = allocb(54);
+        let pkt = unsafe { Packet::<Uninitialized>::wrap(mp1) };
+        assert_eq!(pkt.num_segs(), 1);
+        assert_eq!(pkt.avail(), 54);
+
+        let body = vec![];
+        let mut tcp = TcpHdr::new(3839, 80);
+        tcp.set_seq(4224936861);
```

Comment:
Out of curiosity, how do you pick these sequence numbers? Why not start at 0 or 1?

Reply:
Just pulled it from random tcp dump laying around. The ISN (initial sequence number) is not meant to start at 0/1, or at least I believe it is highly unlikely.

Reply:
Interesting, thank you. Some sources seem to say the ISN _cannot_ be zero, but [the spec](https://www.ietf.org/archive/id/draft-ietf-tcpm-rfc793bis-25.html#section-3.4-21) just says that it must not be predictable, and should be a pseudorandom value based on the timestamp and connection 4-tuple (plus nonce). TIL!